prime-router/src/main/kotlin/azure/BatchDeciderFunction.kt

```diff
@@ -0,0 +1,63 @@
+package gov.cdc.prime.router.azure
+
+import com.microsoft.azure.functions.ExecutionContext
+import com.microsoft.azure.functions.annotation.FunctionName
+import com.microsoft.azure.functions.annotation.StorageAccount
+import com.microsoft.azure.functions.annotation.TimerTrigger
+import gov.cdc.prime.router.cli.tests.CoolTest.Companion.settings
+import java.util.logging.Level
+import kotlin.math.ceil
+import kotlin.math.roundToInt
+
+const val batchDecider = "batchDecider"
+
+/**
+ * This runs as a cron job every minute to determine which receivers, if any, should have batch queue message(s)
+ * added to the stack.
+ */
+class BatchDeciderFunction {
+    @FunctionName(batchDecider)
+    @StorageAccount("AzureWebJobsStorage")
+    fun run(
+        // run every minute (NCRONTAB expression {second} {minute} {hour} {day} {month} {day-of-week})
+        @TimerTrigger(name = "batchDecider", schedule = "0 * * * * *")
+        timerInfo: String,
+        context: ExecutionContext,
+    ) {
+        context.logger.info("$batchDecider: Starting")
+        try {
+            val db = WorkflowEngine().db
+            db.transact { txn ->
+                // TODO: if for some reason we start missing batch runs, we would want to pull all receivers with
+                //  BATCH records that have next_action_at in the past and merge those with the batchInPrevious60Seconds
+                // find all receivers that should have batched within the last 60 seconds
+                settings.receivers.filter { it.timing != null && it.timing.batchInPrevious60Seconds() }
+                    // any that should have batched in the last 60 seconds, get count of outstanding BATCH records
+                    //  (how many actions with BATCH for receiver
+                    .forEach { rec ->
+                        // TODO: to support sending empty batches, we will need to check if the receiver wants that
+                        //  and add the queue message regardless. Currently functionality only adds a message if
+                        //  there is at least one batch to run
+                        // get the number of messages outstanding for this receiver
+                        val recordsToBatch = db.fetchNumberOutstandingBatchRecords(rec.fullName, txn)
+                        val queueMessages = ceil((recordsToBatch.toDouble() / rec.timing!!.maxReportCount.toDouble()))
+                            .roundToInt()
+                        context.logger.info(
```

Comment:
@jimduff-usds @whytheplatypus 
This used to be like this - Gage specifically made a comment to request that we log when there are 0 messages to batch so that we have a log *somewhere* that batch decided did in fact determine that it was time for a specific receiver to batch.  In prod, this will likely not trigger that much since most of our batching is once per day or so.